src/components/entity/state-info.ts

```diff
@@ -0,0 +1,158 @@
+import "@polymer/paper-tooltip/paper-tooltip";
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  LitElement,
+  property,
+  TemplateResult,
+} from "lit-element";
+import { HassEntity } from "home-assistant-js-websocket";
+
+import { computeStateName } from "../../common/entity/compute_state_name";
+import { computeRTL } from "../../common/util/compute_rtl";
+import { HomeAssistant } from "../../types";
+
+import "../ha-relative-time";
+import "./state-badge";
+
+@customElement("state-info")
+class StateInfo extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ attribute: false }) public stateObj?: HassEntity;
+
+  @property({ type: Boolean }) public inDialog = false;
+
+  // property used only in css
+  @property({ type: Boolean, reflect: true }) public rtl = false;
+
+  protected render(): TemplateResult {
+    if (!this.hass || !this.stateObj) {
+      return html``;
+    }
+
+    return html` <state-badge
+        .stateObj=${this.stateObj}
+        .stateColor=${true}
```

Comment:
we were not passing true for `stateColor` which is why the icon was not being colored